sdk/core/Azure.Core/src/RequestOptions.cs

```diff
@@ -0,0 +1,144 @@
+    /// </summary>
+    public class RequestOptions
+    {
+        private bool _frozen;
+
+        private (int Status, bool IsError)[]? _statusCodes;
+        internal (int Status, bool IsError)[]? StatusCodes => _statusCodes;
+
+        private ResponseClassificationHandler[]? _handlers;
+        internal ResponseClassificationHandler[]? Handlers => _handlers;
+
+        internal List<(HttpPipelinePosition Position, HttpPipelinePolicy Policy)>? Policies { get; private set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RequestOptions"/> class.
+        /// </summary>
+        public RequestOptions()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RequestOptions"/> class
+        /// that is a copy of the passed-in options.
+        /// <paramref name="options">The RequestOptions to copy.</paramref>
+        /// </summary>
+        public RequestOptions(RequestOptions options)
+        {
+            _frozen = options._frozen;
+            _statusCodes = options._statusCodes;
+            _handlers = options._handlers;
+            Policies = options.Policies;
+            ErrorOptions = options.ErrorOptions;
+        }
+
+        /// <summary>
+        /// Controls under what conditions the operation raises an exception if the underlying response indicates a failure.
+        /// </summary>
+        public ErrorOptions ErrorOptions { get; set; } = ErrorOptions.Default;
+
+        /// <summary>
+        /// Adds an <see cref="HttpPipelinePolicy"/> into the pipeline for the duration of this request.
+        /// The position of policy in the pipeline is controlled by <paramref name="position"/> parameter.
+        /// If you want the policy to execute once per client request use <see cref="HttpPipelinePosition.PerCall"/>
+        /// otherwise use <see cref="HttpPipelinePosition.PerRetry"/> to run the policy for every retry.
+        /// </summary>
+        /// <param name="policy">The <see cref="HttpPipelinePolicy"/> instance to be added to the pipeline.</param>
+        /// <param name="position">The position of the policy in the pipeline.</param>
+        public void AddPolicy(HttpPipelinePolicy policy, HttpPipelinePosition position)
+        {
+            Policies ??= new();
+            Policies.Add((position, policy));
+        }
+
+        /// <summary>
+        /// Customizes the <see cref="ResponseClassifier"/> for this operation.
+        /// Adding a classifier changes the default classification behavior so that it considers
+        /// the passed-in status code to be an error or not, as specified.
+        /// This is useful for cases where you'd like to prevent specific response status codes from being treated as errors by
+        /// logging and distributed tracing policies -- that is, if a response is not classified as an error, it will not appear as an error in
+        /// logs or distributed traces.
```

Comment:
```suggestion
        /// Customizes the <see cref="ResponseClassifier"/> for this operation to
        /// change the default <see cref="Response" /> classification behavior so that it considers
        /// the passed-in status code to be an error or not, as specified.
        /// This is useful for cases where you'd like to prevent specific response status codes from being treated as errors by
        /// logging and distributed tracing policies -- that is, if a response is not classified as an error, it will not appear as an error in
        /// logs or distributed traces.
```